library/std/src/f16/div_euclid.rs

```diff
@@ -0,0 +1,111 @@
+#[cfg(test)]
+mod tests;
+
+/// Software implementation of `f16::div_euclid`.
+#[allow(dead_code)] // f16::div_euclid is cfg(not(test))
+pub(crate) fn div_euclid(a: f16, b: f16) -> f16 {
+    if let Some((a_neg, a_exp, a_m)) = normal_form(a)
+        && let Some((b_neg, b_exp, b_m)) = normal_form(b)
+    {
+        let exp = a_exp - b_exp;
+        match (a_neg, b_neg) {
+            (false, false) => div_floor(exp, a_m, b_m),
+            (true, false) => -div_ceil(exp, a_m, b_m),
+            (false, true) => -div_floor(exp, a_m, b_m),
+            (true, true) => div_ceil(exp, a_m, b_m),
+        }
+    } else {
+        // `a` or `b` are +-0.0 or infinity or NaN.
+        // `a / b` is also +-0.0 or infinity or NaN.
+        // There is no need to round to an integer.
+        a / b
+    }
+}
```

Comment:
I observe this discrepancy:

```rust
div_euclid(-15.625, 0.00013637542724609375) => -49024.0
exact_div_euclid(-15.625, 0.00013637542724609375).as_f16() => -inf
exact_div_euclid(-15.625, 0.00013637542724609375) => Some(-114574)
-15.625 / 0.00013637542724609375 => -inf
```

Reply:
The correct answer is -inf, this was caused by the overflow bug in `f16::div_ceil`, fixed.